packages/arb-token-bridge-ui/src/pages/api/withdrawals.ts

```diff
@@ -0,0 +1,138 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql } from '@apollo/client'
+import {
+  FetchWithdrawalsFromSubgraphResult,
+  getL2SubgraphClient
+} from 'token-bridge-sdk'
+
+// Extending the standard NextJs request with Withdrawal-params
+type NextApiRequestWithWithdrawalParams = NextApiRequest & {
+  query: {
+    l2ChainId: string
+    address: string
+    search?: string
+    page?: string
+    pageSize?: string
+    fromBlock?: string
+    toBlock?: string
+  }
+}
+
+type WithdrawalResponse = {
+  data: FetchWithdrawalsFromSubgraphResult[]
+  message?: string // in case of any error
+}
+
+export default async function handler(
+  req: NextApiRequestWithWithdrawalParams,
+  res: NextApiResponse<WithdrawalResponse>
+) {
+  try {
+    const {
+      address,
+      search = '',
+      l2ChainId,
+      page = '0',
+      pageSize = '10',
+      fromBlock,
+      toBlock
+    } = req.query
+
+    // validate method
+    if (req.method !== 'GET') {
+      res
+        .status(400)
+        .send({ message: `invalid_method: ${req.method}`, data: [] })
+      return
+    }
+
+    // validate the request parameters
+    if (!l2ChainId || !address) {
+      const errorMessage = []
+      if (!l2ChainId) errorMessage.push('<l2ChainId> is required')
+      if (!address) errorMessage.push('<address> is required')
+
+      res.status(400).json({
+        message: `incomplete request: ${errorMessage.join(', ')}`,
+        data: []
+      })
+    }
+
+    const subgraphResult = await getL2SubgraphClient(Number(l2ChainId)).query({
+      query: gql`{
+            withdrawals(
+                where: {
+                sender: "${address}",
+                ${
+                  typeof fromBlock !== 'undefined'
+                    ? `l2BlockNum_gte: ${Number(fromBlock)}`
+                    : ''
+                }
+                  ${
+                    typeof toBlock !== 'undefined'
+                      ? `l2BlockNum_lte: ${Number(toBlock)}`
+                      : ''
+                  }  
+                  ${search ? `l2TxHash_contains: "${search}"` : ''}
+                }
+                orderBy: l2BlockTimestamp
+                orderDirection: desc
+                first: ${Number(pageSize)},
+                skip: ${Number(page) * Number(pageSize)}
+            ) {
+                id,
+                type,
+                sender,
+                receiver,
+                ethValue,
+                l1Token {
+                    id
+                },
+                tokenAmount,
+                isClassic,
+                l2BlockTimestamp,
+                l2TxHash,
+                l2BlockNum
+            }
+        }`
+    })
+
+    const transactions: FetchWithdrawalsFromSubgraphResult[] =
+      subgraphResult.data.withdrawals.map((eventData: any) => {
+        const {
```

Comment:
I'm not sure what needs to be achieved here by destructing? Aren't we returning the same object in the end? 

Seems like `return eventData` has the same outcome